pkg/controller/vsphere/reconciler.go

```diff
@@ -296,12 +295,21 @@ func (r *Reconciler) delete() error {
 		}
 	}
 
-	if _, err := vm.powerOffVM(); err != nil {
-		return err
+	powerState, err := vm.getPowerState()
+	if err != nil {
+		return fmt.Errorf("can not determine %v vm power state", r.machine.GetName())
+	}
```

Comment:
Could this prevent us from deleting machines? What kind of errors can occur here?

Reply:
It could.
What can go wrong: 
- Unknown state which does not not match with one of the bunch of constants https://github.com/openshift/machine-api-operator/blob/master/pkg/controller/vsphere/reconciler.go#L923
- permissions issue, in case of no `VirtualMachine.Interact.PowerOff` permission there will be an error i believe.
- `getPowerState` makes a request to vcenter, whole bunch of things might potentially happen there. (500, 503, connectivity issues, etc)

Reply:
>We have accounted for all currently available in the SDK so hopefully that's not the case

Yup, right. But its just a strings down the line. No hard guarantees there.

>This is already a potential problem right? Though not for failed create and then delete.

I was mistaken a bit there. This particular line should require only read permissions. In such case the whole thing will fail a waay before execution will hit this.

> Before this, we are checking if the VM exists, line 258, if there was going to be an error like this, I would expect it to occur there. I think we are safe to proceed here, I don't think this increases the risk/if it does, the increase is minimal

Value from this is using three lines below. I think it's necessary to return here in case of error, even if its low probable .